Equilobe.DailyReport.SL/ReportService.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Equilobe.DailyReport.DAL;
+using Equilobe.DailyReport.Models.Interfaces;
+using Equilobe.DailyReport.Models.Jira;
+using Equilobe.DailyReport.Models.ReportFrame;
+using Equilobe.DailyReport.Models.Storage;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Equilobe.DailyReport.SL
+{
+    public class ReportService : IReportService
+    {
+        public IJiraService JiraService { get; set; }
+
+        #region IJiraService Implementation
+        public void UpdateDashboardData(long instanceId)
+        {
+            using (var db = new ReportsDb())
+            {
+                var users = GetUsersFromInstance(db, instanceId)
```

Comment:
var jiraContexts = db.Settings.select( ... )
var users = GetUsers(jiraContexts )
UpdateUsers()

Reply:
I think this call here is a little problematic, because the connection to the database stays open all time, until the requests to JIRA complete, which might take some time, keeping in mind that there are several users to download. So it would be better to get the necessary data in one request, and then close the db connection, then fetch the users from JIRA, and then open a new connection to the db to update the users.